matminer/featurizers/structure.py

```diff
@@ -1172,34 +1209,44 @@ def from_preset(preset):
             CoordinationNumber from a preset.
         """
         nn = getattr(pmg_le, preset)
-        return BagofBonds(nn())
+        return BagofBonds(nn(), **kwargs)
 
-    def featurize_dataframe(self, df, col_id, *args, **kwargs):
+    def fit(self, structures):
         """
-        Compute features for all entries contained in input dataframe.
-        Necessary for returning the correct unified dataframe.
+        Define the bond types allowed to be returned during each featurization.
+        Bonds found during featurization which are not allowed will be omitted
+        from the returned dataframe or matrix.
+
+        Fit BagofBonds by either passing an iterable of structures to
+        training_data or by defining the bonds explicitly with allowed_bonds
+        in __init__.
 
         Args:
-            df (Pandas dataframe): Dataframe containing input data
-            col_id (str or [str]): The dataframe key corresponding to structures
+            structures (Series/list): An iterable of pymatgen Structure
+                objects which will be used to determine the allowed bond
+                types.
 
         Returns:
-            (DataFrame) BagofBonds-featurized dataframe
+            None
         """
+        listlike = (tuple, list, np.ndarray, pd.Series)
+        if not isinstance(structures, listlike):
+            structures = [structures]
```

Comment:
don't fix it for the user. Otherwise, the user might try to send a single Structure into fit() for this featurizer, have it work out, and think that's an OK usage of the code. Just throw an error if they give the wrong argument. Otherwise *every* featurizer should fix Structure -> [Structures], and I don't think we should do that.

Reply:
I will just merge and change that, it's probably the easiest